pkg/provisioner/ironic/ironic.go

```diff
@@ -234,6 +234,14 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 	if ironicNode == nil {
 		p.log.Info("registering host in ironic")
 
+		properties := map[string]interface{}{}
+
+		if p.host.Spec.BootMode == metal3v1alpha1.Legacy {
+			properties["capabilities"] = "boot_mode:bios"
+		} else if p.host.Spec.BootMode == metal3v1alpha1.UEFI {
```

Comment:
> see code in pkg/provisioner/ironic/ironic.go ... it just sets it to bios if legacy is specified. In all the other cases, it is set to uefi
If I am not mistaken, this code should just be :
                if p.host.Spec.BootMode == metal3v1alpha1.Legacy {
			properties["capabilities"] = "boot_mode:bios"
		} else {
                        properties["capabilities"] = "boot_mode:uefi"
                }
The way it is implemented now, does not favor a default.

Reply:
:+1: That's how I'd expect it to be

Reply:
Oh sorry my memory failed there, the issue is related to an initial change i did. It was defaulting to UEFI when i wrote, but I got suggestion from @dhellmann mentioning about not changing the current behaviour. So i removed the default stuff... but not from all comments.
So shall we go without a default value, and do not set anything if boot mode is not set? Or do we want to add a default to UEFI?

Reply:
@bfournie could you please take a look? We should align with how Ironic handles boot mode IMO.

Reply:
The issue is that on a system that supports *neither* BIOS nor UEFI (e.g. because it's OpenFirmware-based), defaulting to either of those is at best weird and at worst might break it.
We do at least need to document what the default Ironic will use on BIOS/UEFI systems is though, because the answer can't be "read through the Ironic source code to find out".

Reply:
Ok -- so I think it's fine to not surprise anyone by not always setting the boot_capabilities fields, and only set if they explicitly set bootMode.

How about we just change the docs then? Mention there's no default.

Reply:
Agree with the documentation comment. 
This seems to be is a case where there needs to be a 3rd value "Neither/Other" and not have any defaults.

Reply:
"Mention there's no default" == "read through the Ironic source code to find out the default"

Reply:
Looking at https://docs.openstack.org/ironic/train/install/advanced.html#boot-mode-support ... seems that default is bios, but it depends on the drivers and ironic configuration...

Reply:
Per https://github.com/openstack/ironic/blob/master/ironic/conf/deploy.py#L91, the current boot_mode default is legacy (BIOS), I think this should be used as the default here.

Reply:
> Looking at https://docs.openstack.org/ironic/train/install/advanced.html#boot-mode-support ... seems that default is bios, but it depends on the drivers and ironic configuration...

As with all of the API discussions here, it doesn't matter what Ironic does. Ironic is an implementation detail. We need to set metal3's default, and then implement that, even if it means always passing something explicitly to Ironic to ensure we get the behavior we want from it. I think that's what you're doing here now, but I want to stress the point because we should not be basing the metal3 API on Ironic's API.